### PR TITLE
5988 components coinmarketcal

### PR DESCRIPTION
**Pull request by luancazarine**
Triggers:
 - Event Added

Note: There's no way to create 'Event Updated' beacause it doesn't have updated_date or similar.

Actions:
 - Search Events
 - Get Coins
 - Get Categories

Note: There's no way to create "Get Event Details" because it doesn't have a GET method filtering by ID.
 
### <samp>🤖 Generated by Copilot at ba09b76</samp>

This pull request adds the `coinmarketcal` app and its actions and sources to the Pipedream repository. The app integrates with the CoinMarketCal API, a service that provides information about cryptocurrency events. The actions allow users to fetch event categories, coins, and events based on various filters. The source emits new events when they are created in CoinMarketCal. The pull request also updates the `pnpm-lock.yaml` file and the `package.json` file for the `coinmarketcal` component.

### <samp>🤖 Generated by Copilot at ba09b76</samp>

> _To work with the CoinMarketCal API_
> _We added an app and some actions, you see_
> _With `listCoins` and `searchEvents`_
> _And `listCategories` for reference_
> _And a source for new events called `event-added`_

### <samp>🤖 Generated by Copilot at ba09b76</samp>

*  Add `coinmarketcal` app and its actions and sources to integrate with the CoinMarketCal API ([link](https://github.com/PipedreamHQ/pipedream/pull/6002/files?diff=unified&w=0#diff-6014a03f8fe32189eb8e739fce49df9098435fd5e7d209204a3f0eb40e533778R1-R24), [link](https://github.com/PipedreamHQ/pipedream/pull/6002/files?diff=unified&w=0#diff-7e1522d5794fffdb0af19d253c9619bad0d357dc43a9287af181cd983fc83279R1-R20), [link](https://github.com/PipedreamHQ/pipedream/pull/6002/files?diff=unified&w=0#diff-90f713acbf37b58c8ec9906e60ffb63a1f6cbeaa96544a712fa06f22d04d8cf5R1-R20), [link](https://github.com/PipedreamHQ/pipedream/pull/6002/files?diff=unified&w=0#diff-8a8bd6e6fb50226d6d203750947f4e3d7024fdb7c02600e5351305e5f5cebcd5R1-R103), [link](https://github.com/PipedreamHQ/pipedream/pull/6002/files?diff=unified&w=0#diff-ecfb1f2dd1eecd4ce567f93000efb29f9bf3a3e2bf830acf4661eeddab8c2733R1-R101), [link](https://github.com/PipedreamHQ/pipedream/pull/6002/files?diff=unified&w=0#diff-6157700721231f19b6a95033acdbffeea3ed1cf0ff9ecff9589859c113ae1fcfR1-R88))
  * Define the `coinmarketcal` app in `coinmarketcal.app.mjs` with methods for listing categories, coins, and events, and prop definitions for authentication and pagination ([link](https://github.com/PipedreamHQ/pipedream/pull/6002/files?diff=unified&w=0#diff-ecfb1f2dd1eecd4ce567f93000efb29f9bf3a3e2bf830acf4661eeddab8c2733R1-R101))
  * Define the `coinmarketcal-list-categories` action in `list-categories.mjs` to fetch and return all event categories ([link](https://github.com/PipedreamHQ/pipedream/pull/6002/files?diff=unified&w=0#diff-7e1522d5794fffdb0af19d253c9619bad0d357dc43a9287af181cd983fc83279R1-R20))
  * Define the `coinmarketcal-list-coins` action in `list-coins.mjs` to fetch and return all coins ([link](https://github.com/PipedreamHQ/pipedream/pull/6002/files?diff=unified&w=0#diff-90f713acbf37b58c8ec9906e60ffb63a1f6cbeaa96544a712fa06f22d04d8cf5R1-R20))
  * Define the `coinmarketcal-search-events` action in `search-events.mjs` to query and return events based on specified filters, and export a summary of the results ([link](https://github.com/PipedreamHQ/pipedream/pull/6002/files?diff=unified&w=0#diff-8a8bd6e6fb50226d6d203750947f4e3d7024fdb7c02600e5351305e5f5cebcd5R1-R103))
  * Define the `coinmarketcal-event-added` source in `event-added.mjs` to emit new events created in CoinMarketCal, and generate a summary and an ID for each event ([link](https://github.com/PipedreamHQ/pipedream/pull/6002/files?diff=unified&w=0#diff-6157700721231f19b6a95033acdbffeea3ed1cf0ff9ecff9589859c113ae1fcfR1-R88))
  * Add `constants.mjs` to export arrays of possible values for some filters used by the `coinmarketcal-search-events` action ([link](https://github.com/PipedreamHQ/pipedream/pull/6002/files?diff=unified&w=0#diff-6014a03f8fe32189eb8e739fce49df9098435fd5e7d209204a3f0eb40e533778R1-R24))
  * Update the version and main entry point in `package.json` to reflect the changes ([link](https://github.com/PipedreamHQ/pipedream/pull/6002/files?diff=unified&w=0#diff-7c62a5b7b8b1afbc8c5825242cd0ac04154551637502576a0225dfcc7121184bL3-R11))
* Add `components/coinmarketcal_demo_app` package as a dependency in `pnpm-lock.yaml` to register the demo app that uses the `coinmarketcal` app and its actions and sources ([link](https://github.com/PipedreamHQ/pipedream/pull/6002/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR770-R772))
* Resolve conflicts with other pull requests that added `components/mindmeister` and `components/omniconvert` packages as dependencies in `pnpm-lock.yaml` ([link](https://github.com/PipedreamHQ/pipedream/pull/6002/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR2341-R2343), [link](https://github.com/PipedreamHQ/pipedream/pull/6002/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR2605-R2610))
* Delete unused files `components/coinmarketcal/.gitignore` and `components/coinmarketcal/app/coinmarketcal.app.ts` ([link](https://github.com/PipedreamHQ/pipedream/pull/6002/files?diff=unified&w=0#diff-905e56a4148457e29e50ec6f9b078c8c5ed800d7d4e1ad27c1aeca947465d1f1), [link](https://github.com/PipedreamHQ/pipedream/pull/6002/files?diff=unified&w=0#diff-bf718f96e2f998916f8177ee1beb8a038da17f22ae9714ea0dfa49a11bb5dece))

### CODE REVIEW

File: components/coinmarketcal/actions/list-categories/list-categories.mjs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  type: "action",
+  props: {
+    coinmarketcal,
+  },
+  async run({ $ }) {
+    const response = await this.coinmarketcal.listCategories({
+      $,
+    });
+
+    $.export("$summary", "Categories were successfully fetched!");
```
Comment (vellames) [inline components/coinmarketcal/actions/list-categories/list-categories.mjs:17]: Could you please add the total of categories fetched?

**Review by vellames — CHANGES_REQUESTED** [review]

File: components/coinmarketcal/actions/list-categories/list-categories.mjs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  type: "action",
+  props: {
+    coinmarketcal,
+  },
+  async run({ $ }) {
+    const response = await this.coinmarketcal.listCategories({
+      $,
+    });
+
+    $.export("$summary", `${response.body.length} categories were successfully fetched!`);
```
Comment (vellames) [inline components/coinmarketcal/actions/list-categories/list-categories.mjs:17]: Can you handle the case with one as the length as well, please?

File: components/coinmarketcal/actions/list-coins/list-coins.mjs [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  type: "action",
+  props: {
+    coinmarketcal,
+  },
+  async run({ $ }) {
+    const response = await this.coinmarketcal.listCoins({
+      $,
+    });
+
+    $.export("$summary", `${response.body.length} coins were successfully fetched!`);
```
Comment (vellames) [inline components/coinmarketcal/actions/list-coins/list-coins.mjs:17]: Can you handle the case with one as the length as well, please?

File: components/coinmarketcal/actions/search-events/search-events.mjs [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+import {
+  SHOW_ONLY,
+  SORT_BY,
+  TRANSLATIONS,
+} from "../common/constants.mjs";
+
+export default {
+  key: "coinmarketcal-search-events",
+  name: "Search Events",
+  version: "0.0.5",
```
Comment (vellames) [inline components/coinmarketcal/actions/search-events/search-events.mjs:11]:
```suggestion
  version: "0.0.1",
```

**Review by vellames — CHANGES_REQUESTED** [review]

**Review by vellames — APPROVED** [review]

**Comment by vunguyenhung** [comment]
Hi everyone, all test cases are passed! Ready for release! 

Test report
[CoinMarketCal_5988_1311.pdf](https://github.com/PipedreamHQ/pipedream/files/11246754/CoinMarketCal_5988_1311.pdf)

**Pull request merged**